### PR TITLE
Use the Windows Multimedia Timer

### PR DESCRIPTION
**Pull request by sgeto**

### CODE REVIEW

**Comment by guyharris** [comment]
Might as well just unconditionally use it on Windows regardless of what build system we use.  Done in eedd51f5528535b2914e3f089db1aed0b642cda4.

**Comment by sgeto** [comment]
okay but the multimedia timer depends on `winmm `and all of tcpdump on `ws2_32`

File: configure.in [hunk 1]
```diff
@@ -90,6 +90,11 @@ darwin*)
 		esac
 	fi
 	;;
+*mingw*)
+	AC_DEFINE(USE_WIN32_MM_TIMER, 1,
+		[Define to 1 to use the Windows Multimedia Timer])
+	LIBS="$LIBS -lws2_32 -lwinmm"
```
Comment (sgeto) [inline configure.in]: Here

**Comment by guyharris** [comment]
> okay but the multimedia timer depends on `winmm`

So what about using `CreateTimerQueueTimer()`?

**Comment by guyharris** [comment]
> So what about using `CreateTimerQueueTimer()`?

Done in bfb7669b376344e4a5f0459138284cc305678ccd.

**Comment by sgeto** [comment]
Hehe. You're fast...
What about winsock2. New PR?

**Comment by sgeto** [comment]
Well that didn't work:

```
[Ali:~/projects/tcpdump] $ make
gcc -ffloat-store -DHAVE_CONFIG_H   -D_U_="__attribute__((unused))" -I. -IC:/msys64/mingw64/include  -I/usr/inet6/include -g -O2 -c ./tcpdump.c
./tcpdump.c:228:34: error: 'INVALID_HANDLE' undeclared here (not in a function); did you mean 'WSA_INVALID_HANDLE'?
     static HANDLE timer_handle = INVALID_HANDLE;
                                  ^~~~~~~~~~~~~~
                                  WSA_INVALID_HANDLE
In file included from C:/msys64/mingw64/x86_64-w64-mingw32/include/minwindef.h:163:0,
                 from C:/msys64/mingw64/x86_64-w64-mingw32/include/windef.h:8,
                 from C:/msys64/mingw64/x86_64-w64-mingw32/include/windows.h:69,
                 from C:/msys64/mingw64/x86_64-w64-mingw32/include/winsock2.h:23,
                 from ./netdissect-stdinc.h:163,
                 from ./tcpdump.c:58:
./tcpdump.c: In function 'main':
./tcpdump.c:2216:7: error: expected ')' before numeric constant
       WT_EXECUTEDEFAULT|WT_EXECUTELONGFUNCTION);
       ^
./tcpdump.c:2214:3: error: too few arguments to function 'CreateTimerQueueTimer'
   CreateTimerQueueTimer(&timer_handle, NULL,
   ^~~~~~~~~~~~~~~~~~~~~
In file included from C:/msys64/mingw64/x86_64-w64-mingw32/include/winbase.h:39:0,
                 from C:/msys64/mingw64/x86_64-w64-mingw32/include/windows.h:70,
                 from C:/msys64/mingw64/x86_64-w64-mingw32/include/winsock2.h:23,
                 from ./netdissect-stdinc.h:163,
                 from ./tcpdump.c:58:
C:/msys64/mingw64/x86_64-w64-mingw32/include/threadpoollegacyapiset.h:18:29: note: declared here
   WINBASEAPI WINBOOL WINAPI CreateTimerQueueTimer (PHANDLE phNewTimer, HANDLE TimerQueue, WAITORTIMERCALLBACK Callback, PVOID Parameter, DWORD DueTime, DWORD Period, ULONG Flags);
                             ^~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:71: tcpdump.o] Error 1
[Ali:~/projects/tcpdump] 2 $
```

**Comment by sgeto** [comment]
You need to change INVALID_HANDLE to WSA_INVALID_HANDLE and add a comma after the second int (period) in CreateTimerQueueTimer().

**Comment by guyharris** [comment]
> 'INVALID_HANDLE' undeclared here

568927a2bce300035cf25e9cd387986f4a72b10c

**Comment by guyharris** [comment]
And 8f2cde6936738fc564e47785a9b83a548affa2ad.

**Comment by sgeto** [comment]
Including windows.h directly again :/
It's already there
```
In file included from C:/msys64/mingw64/x86_64-w64-mingw32/include/winbase.h:39:0,
                 from C:/msys64/mingw64/x86_64-w64-mingw32/include/windows.h:70,
                 from C:/msys64/mingw64/x86_64-w64-mingw32/include/winsock2.h:23,
                 from ./netdissect-stdinc.h:163,
                 from ./tcpdump.c:58:
```

**Pull request closed without merging**